src/main/java/com/meilisearch/sdk/Search.java

```diff
@@ -67,7 +69,12 @@ SearchResult search(String uid, String q) throws MeilisearchException {
      * @return search results
      * @throws MeilisearchException Search Exception or Client Error
      */
-    SearchResult search(String uid, SearchRequest sr) throws MeilisearchException {
+    Searchable search(String uid, SearchRequest sr) throws MeilisearchException {
+        if (sr != null
+                && (sr.getPage() != null && sr.getPage() != 0
+                        || sr.getHitsPerPage() != null && sr.getHitsPerPage() != 0)) {
+            return httpClient.jsonHandler.decode(rawSearch(uid, sr), SearchResultPaginated.class);
```

Comment:
```suggestion
        if (sr != null && (sr.getPage() != null || sr.getHitsPerPage() != null)) {
            return httpClient.jsonHandler.decode(rawSearch(uid, sr), SearchResultPaginated.class);
```

Even when you request with no hits and page 0, the result will still be a paginated request like this:

```json
{
    "hits": [],
    "query": "prinec",
    "processingTimeMs": 2,
    "hitsPerPage": 0,
    "page": 0,
    "totalPages": 0,
    "totalHits": 2
}
```